tkg/client/update_cluster_metadata.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	crtclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/tanzu-framework/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/tkg/constants"
+)
+
+const clusterMetadataNamespace = "tkg-system-public"
```

Comment:
It seems this namespace is already removed.

Reply:
In the func `createOrUpdateNamespaceRole` below, it will call `clusterClient.CreateNamespace(clusterMetadataNamespace)` to create this namespace. And this `CreateNamespace` func can also handle the already exists error